test/blurring-method.js

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('blurring method', function() {
+  const testCases = [
+    {
+      'description': 'jpg image',
+      'inputImageUrl': 'test-files/64kb.jpg',
+      'expectedImageUrl': 'blurred/test1.png',
+      'rects': [
+        {
+          'leftX': 0,
+          'topY': 0,
+          'width': 100,
+          'height': 100,
+        },
+        {
+          'leftX': 100,
+          'topY': 100,
+          'width': 50,
+          'height': 10,
+        },
+      ],
+    },
+    {
+      'description': 'png image',
+      'inputImageUrl': 'test-files/537kb.png',
+      'expectedImageUrl': 'blurred/test2.png',
+      'rects': [
+        {
+          'leftX': 0,
+          'topY': 0,
+          'width': 100,
+          'height': 100,
+        },
+        {
+          'leftX': 100,
+          'topY': 100,
+          'width': 50,
+          'height': 10,
+        },
+      ],
+    },
+    {
+      'description': 'intersecting rectangles',
+      'inputImageUrl': 'test-files/64kb.jpg',
+      'expectedImageUrl': 'blurred/test3.png',
+      'rects': [
+        {
+          'leftX': 100,
+          'topY': 100,
+          'width': 400,
+          'height': 100,
+        },
+        {
+          'leftX': 30,
+          'topY': 30,
+          'width': 200,
+          'height': 300,
+        },
+      ],
+    },
+    {
+      'description': 'one big rectangle',
+      'inputImageUrl': 'test-files/64kb.jpg',
+      'expectedImageUrl': 'blurred/test4.png',
+      'rects': [
+        {
+          'leftX': 0,
+          'topY': 0,
+          'width': 615,
+          'height': 370,
+        },
+      ],
+    },
+  ];
+
+  /**
+   * Function to generate tests from test cases above.
+   * @param {Object} testCase
+   */
+  function makeTest(testCase) {
+    it(testCase.description, function() {
+      return new Promise(async function(resolve, reject) {
+        const rectsToBlur = testCase.rects;
+        const inputImage = await getImageFromPath(testCase.inputImageUrl);
+        const expectedImage = await getFileBlob(testCase.expectedImageUrl);
```

Comment:
Curious why you use 2 different functions to get images?

Reply:
The first one returns Image object, and the second returns Blob. I pass inputImage as parameter to getImageWithBlurredAreas function and then compare the output of this function (which is canvas element, but resemblejs can handle it) with expectedImage.

Reply:
Maybe naming it differently, like expectedImageBlob would be more clear?

Reply:
I also use this function in getFileObject function (test/helpers.js file), where it doesn't get expected image blob, but gets the blob of some file passed to this function as parameter. getFileObject function is used in some other files, e.g. in test/image-upload.js file.

Reply:
I will leave a comment explaining why there are different functions.